torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -519,7 +529,10 @@ struct CodeImpl {
   }
 
   void insertInstruction(OpCode op, int64_t X = 0, uint64_t N = 0) {
-    instructions_.emplace_back(op, X, N);
+    instructions_.emplace_back(
+        op,
+        safe_narrow_cast<int32_t, int64_t>(X),
```

Comment:
wait, won't it throw if there are > int32 distinct warnings? it actually can happen because you keep increasing the same counter across all models, so it might overflow eventually. you probably want explicit static_cast in emitWarn i n this case as the truncation is ok

Reply:
Removed this logic since index is no longer needed

Reply:
Added `static_cast` at time of emitting warn op.